verification/cbmc/sources/make_common_data_structures.c

```diff
@@ -30,13 +30,39 @@
 #include <proof_helpers/make_common_data_structures.h>
 #include <proof_helpers/proof_allocators.h>
 
-void ensure_alg_properties_attempt_allocation(struct aws_cryptosdk_alg_properties *const alg_props) {
-    size_t md_name_size;
-    alg_props->md_name = can_fail_malloc(md_name_size);
-    size_t cipher_name_size;
-    alg_props->cipher_name = can_fail_malloc(cipher_name_size);
-    size_t alg_name_size;
-    alg_props->alg_name = can_fail_malloc(alg_name_size);
+const EVP_MD *nondet_EVP_MD_ptr(void);
+const EVP_CIPHER *nondet_EVP_CIPHER_ptr(void);
+
+struct aws_cryptosdk_alg_impl *ensure_impl_attempt_allocation(const size_t max_len) {
```

Comment:
https://github.com/aws/aws-encryption-sdk-c/issues/662

Reply:
Great, I cannot agree more. Sometimes we allocates the top structure itself (impl), and some others only a few members... We should make a distinction.